l1-contracts/src.ts/hyperchain-upgrade.ts

```diff
@@ -181,9 +233,70 @@ async function integrateEraIntoBridgehubAndUpgradeL2SystemContract(deployer: Dep
 }
 
 async function upgradeL2Bridge(deployer: Deployer) {
-  // upgrade L2 bridge contract, we do this directly via the L2
-  // set initializeChainGovernance in L1SharedBridge
-  deployer;
+  // todo deploy l2 bridge here
+  // L2_SHARED_BRIDGE_IMPLEMENTATION_BYTECODE
+  const l2BridgeImplementationAddress = ADDRESS_ONE; // todo
+
+  // upgrade from L1 governance. This has to come from governacne on L1.
+  const l2BridgeAbi = ["function initialize(address, address, bytes32, address)"];
+  const l2BridgeContract = new ethers.Contract(ADDRESS_ONE, l2BridgeAbi, deployer.deployWallet);
+  const l2Bridge = l2BridgeContract.interface; //L2_SHARED_BRIDGE_INTERFACE;
+  const l2BridgeCalldata = l2Bridge.encodeFunctionData("initialize", [
+    deployer.addresses.Bridges.SharedBridgeProxy,
+    deployer.addresses.Bridges.ERC20BridgeProxy,
+    hashL2Bytecode(BEACON_PROXY_BYTECODE),
```

Comment:
just in case, the old (Era) contract supports `function l2TokenAddress`. If we still want to support it (at least specifically for Era), we must not pass a new beacon proxy bytecode hash for the SharedBridge on Era.

While the function was deleted from the L1SharedBridge, it is still present in the `L1ERC20Bridge`

Reply:
Ok, I'll just move the initialization in the L2SharedBridge itself so that we only set the proxyBytecodeHash if we are not on Era.